src/platform/silabs/EndpointQueueFilter.cpp

```diff
@@ -0,0 +1,141 @@
+#include "EndpointQueueFilter.h"
+#include <string.h>
+#include <support/CodeUtils.h>
+#include <support/logging/CHIPLogging.h>
+
+namespace chip {
+namespace Inet {
+
+using FilterOutcome = EndpointQueueFilter::FilterOutcome;
+using namespace chip::Logging;
+
+namespace {
+
+bool IsValidMdnsHostName(const chip::CharSpan & hostName)
+{
+    for (size_t i = 0; i < hostName.size(); ++i)
+    {
+        char ch_data = *(hostName.data() + i);
+        if (!((ch_data >= '0' && ch_data <= '9') || (ch_data >= 'A' && ch_data <= 'F')))
+        {
+            return false;
+        }
+    }
+    return true;
+}
+
+bool PayloadContains(const chip::System::PacketBufferHandle & payload, const chip::ByteSpan & byteSpan)
+{
+    if (payload->HasChainedBuffer() || payload->TotalLength() < byteSpan.size())
+    {
+        return false;
+    }
+    for (size_t i = 0; i <= payload->TotalLength() - byteSpan.size(); ++i)
+    {
+        if (memcmp(payload->Start() + i, byteSpan.data(), byteSpan.size()) == 0)
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+bool PayloadContainsCaseInsensitive(const chip::System::PacketBufferHandle & payload, const ByteSpan & name)
+{
+    ReturnErrorCodeIf(name.size() < HostNameFilter::kHostNameLengthMax, false);
+    uint8_t lower_case[HostNameFilter::kHostNameLengthMax];
+    memcpy(lower_case, name.data(), name.size());
+    for (size_t i = 0; i < sizeof(lower_case); ++i)
+    {
+        if (lower_case[i] <= 'F' && lower_case[i] >= 'A')
+        {
+            lower_case[i] = static_cast<uint8_t>('a' + lower_case[i] - 'A');
+        }
+    }
+    return PayloadContains(payload, name) || PayloadContains(payload, ByteSpan(lower_case));
```

Comment:
This is incorrect --> it will fail to deal with mixed case. Suggest having a single loop of non-case-sensitive search, rather than these odd copies

Reply:
Having only 2 search now, "matter" and "hostname"